packages/datadog-instrumentations/src/google-cloud-pubsub.js

```diff
@@ -151,6 +160,55 @@ addHook({ name: '@google-cloud/pubsub', versions: ['>=1.2'], file: 'build/src/le
   return obj
 })
 
+function injectTraceContext (attributes, pubsub, topicName) {
+  if (attributes['x-datadog-trace-id'] || attributes.traceparent) return
+
+  try {
+    const tracer = require('../../dd-trace')
+    const activeSpan = tracer.scope().active()
+    if (!activeSpan) return
+
+    tracer.inject(activeSpan, 'text_map', attributes)
+
+    const traceIdUpperBits = activeSpan.context()._trace.tags['_dd.p.tid']
+    if (traceIdUpperBits) attributes['_dd.p.tid'] = traceIdUpperBits
+  } catch {
+    // Silently fail - trace context injection is best-effort
```

Comment:
Let's add a log here as well. Otherwise it would be difficult to diagnose if the feature isn't working for customers who expect it to.

I'd also like to know if you're seeing any failures here during test suite runs. For example if the `activeSpan.context()._trace.tags['_dd.p.tid']` line fails with something like "cannot access _trace of undefined" then the code might better be served looking like activeSpan.context()?._trace.tags['_dd.p.tid']`, and nothing else ever fails, then maybe we can remove the try/catch entirely.